src/zarr/storage/_obstore.py

```diff
@@ -186,9 +186,17 @@ def supports_deletes(self) -> bool:
     async def delete(self, key: str) -> None:
         # docstring inherited
         import obstore as obs
+        from obstore.exceptions import NotFoundError
```

Comment:
IMO let's just handle `FileNotFoundError` for parity with existing code:
https://github.com/zarr-developers/zarr-python/blob/2911be8f20357c7f717eed410ccc99782d487808/src/zarr/storage/_obstore.py#L153-L154

I can't get multiple inheritance of exceptions to work in obstore, so it's possible I'll just remove the `NotFoundError` in a future release.